src/gl/canvas2dRe.re

```diff
@@ -64,7 +68,14 @@ module LineJoin: LineJoinType = {
   let miter : t = "miter";
 };
 
+type image 'a =
+  | Number: image float
+  | ImageData: image ImageRe.t;
 
+type style _ =
```

Comment:
I may misunderstand something, but it seems like `style` needs a payload in its constructor, since we have a `fillStyle` getter that returns a valid `style` that we have to unwrap to get a value, right? This has an odd effect on `setFillStyle`. I figured it couldn't be `t => style 'a => 'a` anymore, since they'd have to call it with something like `ctx (String "whatever") "red"`, which seems weird. I converted it to switch on a single `style` value, but I can't seem to make the resulting `setXXX` external call _not_ use a `Block.__(0, ["red"])` in the JS. Will that outright break, or is the generated JS just weirder? It looks like it's still going to resolve to `"red"`, isn't it?